lnwallet/channel.go

```diff
@@ -4351,6 +4347,17 @@ func (lc *LightningChannel) ReceiveHTLC(htlc *lnwire.UpdateAddHTLC) (uint64, err
 		OnionBlob: htlc.OnionBlob[:],
 	}
 
+	localACKedIndex := lc.remoteCommitChain.tail().ourMessageIndex
+
+	// Clamp down on the number of HTLC's we can receive by checking the
+	// commitment sanity.
+	err := lc.validateCommitmentSanity(
```

Comment:
yeah, nice I think this is it. If I'm not mistaken this will also handle the `CASE 2` you outlined earlier, since it will apply the settles/fails accordingly?

Reply:
Yup it applies the same logic that the sender applies so it handles case 2